### PR TITLE
feat: read s12 device info

### PR DESCRIPTION
**Pull request by admons**
## Description

> added the ability to read S12 devices, this is the first step to add support for s12 device (2 shutters and 1 light)

**Related issue (if any):** helps to https://github.com/TomerFi/aioswitcher/issues/585

## Checklist

- [x] I have followed this repository's contributing guidelines.
- [x] I will adhere to the project's code of conduct.

## Additional information

> Now the api support should be developed.

### CODE REVIEW

**Comment by TomerFi** [comment]
@admons Thank you very much!
@thecode would you mind taking a look at this?

**Comment by admons** [comment]
@TomerFi i can help writing the api with some help on how to see the commands that are being sent to the device. Do you have this in writing anywhere?

**Comment by TomerFi** [comment]
> @TomerFi i can help writing the api with some help on how to see the commands that are being sent to the device. Do you have this in writing anywhere?

what api are you referring to?
btw - i'm not familiar with the s12 device, do you have any link for that?

**Comment by admons** [comment]
@TomerFi I'm talking about sending the commands to the device

S12 device is one that has 2 shutters and 1 light - https://switcher.co.il/%D7%9E%D7%95%D7%A6%D7%A8/runner-lights-s12/

What I currently did was to add support for reading the status of the device (what's the current position of the shutters, is the light on/off, is any button is currently being clicked)

What's left is add the commands that we need to send to the device to affect these entities

**Comment by TomerFi** [comment]
@admons 

> @TomerFi I'm talking about sending the commands to the device
> 
> S12 device is one that has 2 shutters and 1 light - https://switcher.co.il/%D7%9E%D7%95%D7%A6%D7%A8/runner-lights-s12/
> 

Thanks, haven't heard of this one. 🙏🏻 

> What I currently did was to add support for reading the status of the device (what's the current position of the shutters, is the light on/off, is any button is currently being clicked)
> 

Cool, let's get this merged than. 😀 
you can fix the _black_ lint issue with either:

```shell
make black-with-fix
```

or:

```shell
poetry run poe black_fix
```

both do the same.

> What's left is add the commands that we need to send to the device to affect these entities
>

It's been a long time since I've last worked on these packets (or the api),
I think @jafar-atili was the last one to work in this section,
maybe he can share some knowledge or pointers about the various packets.

**Comment by admons** [comment]
@TomerFi Made the linting change and pushed

**Comment by TomerFi** [comment]
```shell
poetry run poe lint

poetry run poe isort_fix
```
or

```shell
make lint

make isort-with-fix
```

**Review by TomerFi — APPROVED** [review]

**Comment by TomerFi** [comment]
@admons thank you very much!

**Comment by TomerFi** [comment]
@all-contributors add @admons  for code

File: src/aioswitcher/bridge.py
Comment (thecode) [inline src/aioswitcher/bridge.py:35]: This brakes our current design, we named devices by their function (e.g. Water Heater, Shutter, etc...)
This is a Shutter device (or maybe a dual shutter) but now we name it by the device model

File: src/aioswitcher/bridge.py
Comment (thecode) [inline src/aioswitcher/bridge.py:425]: This would be better to handle with a parameter and keep the standard design, e.g., add option for the number of the controlled shutter, else it would be complicated to add this into the integration since function names needs to be different if we control the first or the 2nd shutter.

File: src/aioswitcher/device/__init__.py
Comment (thecode) [inline src/aioswitcher/device/__init__.py:385]: This is not a category but a device model, does not align with our current design

**Review by thecode — COMMENTED** [review]
@TomerFi I did not have time to review it and it would make sense to wait for a review. I don't think this PR is ready to go.
1. It only add discovery and not control, we should merge it into a branch and not main.
2. Not covered by tests
3. Until now we named devices by their category and not by the device model (see review comments)
4. Most important one, the implementation of this dual shutter will be hard to integrate since we now have 3 different names of fields (single shutter , dual shutter first channel and 2nd channel).
5. Switcher has a device with mixed Light & Shutter we should discuss how we handle such devices before any implementation.

I have told @admons before he started I don't have time to work on it, this is kinda forcing me a review now.

**Comment by TomerFi** [comment]
> @TomerFi I did not have time to review it and it would make sense to wait for a review. I don't think this PR is ready to go.
> 
> 1. It only add discovery and not control, we should merge it into a branch and not main.
> 2. Not covered by tests
> 3. Until now we named devices by their category and not by the device model (see review comments)
> 4. Most important one, the implementation of this dual shutter will be hard to integrate since we now have 3 different names of fields (single shutter , dual shutter first channel and 2nd channel).
> 5. Switcher has a device with mixed Light & Shutter we should discuss how we handle such devices before any implementation.
> 
> I have told @admons before he started I don't have time to work on it, this is kinda forcing me a review now.

Dear @thecode I tagged you 3 days ago and asked if you can review this.
You approved the CI to run but you:
- didn't comment at the conversation at all, until now of course
- didn't assigned yourself
- didn't marked this PR as draft
- didn't contact me in any way, shape, or form, although you have many ways to contact me online, plus my phone number

Now, as soon as I merge this, suddanly you have a lot to say.
And part of what you're saying is that you were aware of this work request prior to pr,
yet it didn't occur to you it might be a good idea to say something, or at the very least mark this pr as draft so other maintainers, such as myself, would know not to merge this.

Shay - I'm not forcing you to review anything!
This is an open-source project in which you are a maintainer, but not obligated to in any way.
If you can, and want to, whenever you have the time, any review or other contribution you'll make as a maintainer will be very much appreciated,
in the same manner, if you do not have the time to do so, it's totally understood.

Please stop shooting at me at every chance you get, this is not the first time I had to tell you how unacceptable this is.

**Comment by TomerFi** [comment]
@admons
I moved this work to a designated branch, [add-s12-support](https://github.com/TomerFi/aioswitcher/tree/add-s12-support) as @thecode suggested.
This work is no longer on `dev` branch, please pull again if needed.

**Pull request merged**
